lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexAggregateFuture.java

```diff
@@ -93,6 +86,10 @@ public synchronized AggregateResult get(final long timeout,
     return result;
   }
 
+  private synchronized boolean taskletIdInputMapIsEmpty() {
+    return taskletIdInputMap.isEmpty();
```

Comment:
Adding `@GuardedBy` annotation to `taskletIdInputMap` will let us know that this object should be synchronized properly.